### PR TITLE
Split into multiple files

### PR DESCRIPTION
**Pull request by ehmicky**
This PR splits `index.js` and `test.js` into multiple files.

Unfortunately, `git diff` is not great at showing file splits, but the lines were just copied, with no change in logic nor behavior.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
 	},
 	"files": [
 		"index.js",
-		"index.d.ts"
+		"index.d.ts",
+		"lib"
```
Comment (ehmicky) [inline package.json:25]: :arrow_up:

**Comment by sindresorhus** [comment]
Can you call it `source` instead of `lib` and move `index.js` and `index.d.ts` into it too?

File: package.json [hunk 2]
```diff
@@ -11,17 +11,17 @@
...
 	"scripts": {
 		"benchmark": "node benchmarks/index.js",
-		"test": "xo && ava && tsd"
+		"test": "xo && ava && tsd --typings source/index.d.ts"
 	},
 	"files": [
-		"index.js",
-		"index.d.ts"
+		"source",
+		"!*.test-d.ts"
```
Comment (ehmicky) [inline package.json:24]: `npm publish --dry`:

```
license               
package.json          
readme.md             
source/array-buffer.js
source/array.js       
source/buffer.js      
source/contents.js    
source/index.d.ts     
source/index.js       
source/string.js      
source/utils.js       
```

**Comment by ehmicky** [comment]
Fixed in 18eabf582531a380265f5d7d11d1fed846b8f6ac :+1:

**Pull request merged**